cognite_toolkit/_cdf_tk/client/api/agents/agents.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import annotations
+
+from collections.abc import Iterator, Sequence
+from typing import overload
+
+from cognite.client._api_client import APIClient
+from cognite.client.utils.useful_types import SequenceNotStr
+
+from cognite_toolkit._cdf_tk.client.data_classes.agents import Agent, AgentList, AgentWrite
+
+
+class AgentsAPI(APIClient):
+    _RESOURCE_PATH = "/ai/agents"
+
+    @overload
+    def __call__(self) -> Iterator[Agent]: ...
+
+    @overload
+    def __call__(self, chunk_size: int) -> Iterator[AgentList]: ...
+
+    def __call__(self, chunk_size: int | None = None) -> Iterator[Agent] | Iterator[AgentList]:
+        """Iterate over agents.
+
+        Args:
+            chunk_size: The number of agents to return in each chunk. None will return all agents.
+
+        Yields:
+            Agent or AgentList
+
+        """
+        return self._list_generator(method="GET", resource_cls=Agent, list_cls=AgentList, chunk_size=chunk_size)
+
+    def __iter__(self) -> Iterator[Agent]:
+        return self.__call__()
+
+    @overload
+    def apply(self, agent: AgentWrite) -> Agent: ...
+
+    @overload
+    def apply(self, agent: Sequence[AgentWrite]) -> AgentList: ...
+
+    def apply(self, agent: AgentWrite | Sequence[AgentWrite]) -> Agent | AgentList:
+        """Create a new agent.
+
+        Args:
+            agent: AgentWrite or list of AgentWrite.
+
+        Returns:
+            Agent object.
+
+        """
+        headers = {"cdf-version": "alpha"}
+        return self._create_multiple(
+            list_cls=AgentList,
+            resource_cls=Agent,
+            items=agent,
+            input_resource_cls=AgentWrite,
+            headers=headers,
+        )
+
+    def retrieve(self, external_id: str | SequenceNotStr[str]) -> AgentList:
```

Comment:
In the PySDK, we use overload to return `Agent | AgenList` depending on whether it is a sequence of strings or a single strings. Since this is only used by Toolkit, we always have sequence of strings, so suggest the following.
```suggestion
    def retrieve(self, external_id: SequenceNotStr[str]) -> AgentList:
```